src/test/java/com/algolia/search/integration/insights/InsightsTest.java

```diff
@@ -0,0 +1,77 @@
+package com.algolia.search.integration.insights;
+
+import com.algolia.search.InsightsClient;
+import com.algolia.search.SearchClient;
+import com.algolia.search.SearchIndex;
+import com.algolia.search.UserInsightsClient;
+import com.algolia.search.integration.TestHelpers;
+import com.algolia.search.integration.models.AlgoliaObject;
+import com.algolia.search.models.indexing.Query;
+import com.algolia.search.models.indexing.SearchResult;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+import org.junit.jupiter.api.Test;
+
+public abstract class InsightsTest {
+  protected final SearchClient searchClient;
+  protected final InsightsClient insightsClient;
+
+  protected InsightsTest(SearchClient searchClient, InsightsClient insightsClient) {
+    this.searchClient = searchClient;
+    this.insightsClient = insightsClient;
+  }
+
+  @Test
+  void testInsights() throws ExecutionException, InterruptedException {
```

Comment:
Nothing to assert? :thinking:

Reply:
It's how we wrote the test for the CTS. The insights API is mostly fire&forget API if we don't get a `400` from the API it's OK because we can't retrieve what was sent to check if it's the same on the server side (as we do for settings for example)